src/legacy/core_plugins/metrics/public/plugin.ts

```diff
@@ -17,14 +17,15 @@
  * under the License.
  */
 
+import { CoreSetup, PluginsServiceSetup } from 'src/core/server';
```

Comment:
I'm not sure if `PluginsServiceSetup` is what you are looking for... This is an interface that you as the plugin author should define, since you know what your plugin dependencies will be.

There's an example in the [migration guide](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md), but something like this:
```typescript
// note that even though this is private, we'd need to export here so we can use it in the `setup.ts` shim
export interface TsvbSetupPlugins {
  data: DataSetup; // import these from `core_plugins/plugin_name/public`
  visualizations: VisualizationsSetup;
}

public setup(core: CoreSetup, plugins: TsvbSetupPlugins) {}
```

Reply:
Nice catch! Added